crates/eips/src/eip7594/sidecar.rs

```diff
@@ -0,0 +1,285 @@
+use crate::{
+    eip4844::{
+        kzg_to_versioned_hash, Blob, BlobAndProofV2, Bytes48, BYTES_PER_BLOB, BYTES_PER_COMMITMENT,
+        BYTES_PER_PROOF,
+    },
+    eip7594::{CELLS_PER_EXT_BLOB, EIP_7594_WRAPPER_VERSION},
+};
+use alloc::{boxed::Box, vec::Vec};
+use alloy_primitives::B256;
+use alloy_rlp::{BufMut, Decodable, Encodable, Header};
+
+#[cfg(feature = "kzg")]
+use crate::eip4844::BlobTransactionValidationError;
+
+/// This represents a set of blobs, and its corresponding commitments and cell proofs.
+///
+/// This type encodes and decodes the fields without an rlp header.
+#[derive(Clone, Default, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
+pub struct BlobTransactionSidecarEip7594 {
+    /// The blob data.
+    #[cfg_attr(
+        all(debug_assertions, feature = "serde"),
+        serde(deserialize_with = "crate::eip4844::deserialize_blobs")
+    )]
+    pub blobs: Vec<Blob>,
+    /// The blob commitments.
+    pub commitments: Vec<Bytes48>,
+    /// List of cell proofs for all blobs in the sidecar, including the proofs for the extension
+    /// indices, for a total of `CELLS_PER_EXT_BLOB` proofs per blob (`CELLS_PER_EXT_BLOB` is the
+    /// number of cells for an extended blob, defined in
+    /// [the consensus specs](https://github.com/ethereum/consensus-specs/tree/9d377fd53d029536e57cfda1a4d2c700c59f86bf/specs/fulu/polynomial-commitments-sampling.md#cells))
+    pub cell_proofs: Vec<Bytes48>,
```

Comment:
we renamed cell_proofs to proofs in blobsbundlev2

I personally prefer cell_proofs here, because this makes it easier to distinguish the two sidecar types, but I wonder if we should rename this here as well, if this type will be used in jsonrpc?
though we could rename via serde name later

Reply:
would like to keep `cell_proofs` too. mb the other PR should've been a simple serde rename as well